apps/api/v2/src/ee/bookings/controllers/bookings.controller.ts

```diff
@@ -34,7 +44,50 @@ import { ApiResponse } from "@calcom/platform-types";
 export class BookingsController {
   private readonly logger = new Logger("ee bookings controller");
 
-  constructor(private readonly oAuthFlowService: OAuthFlowService) {}
+  constructor(
+    private readonly oAuthFlowService: OAuthFlowService,
+    private readonly prismaReadService: PrismaReadService
+  ) {}
+
+  // note(Rajiv): currently this endpoint is atoms only
+  @Get("/")
+  @UseGuards(AccessTokenGuard)
+  async getBookings(
+    @GetUser() user: User,
+    @Query() queryParams: GetBookingsInput
+  ): Promise<ApiResponse<unknown>> {
+    const { filters, cursor, limit } = queryParams;
+    const bookings = await getAllUserBookings({
+      bookingListingByStatus: filters.status,
+      skip: cursor ?? 0,
+      take: limit ?? 10,
+      filters,
+      ctx: {
+        user: { email: user.email, id: user.id },
+        prisma: this.prismaReadService.prisma as unknown as PrismaClient,
+      },
+    });
+
+    return {
+      status: SUCCESS_STATUS,
+      data: bookings,
+    };
+  }
+
+  // note(Rajiv): currently this endpoint is atoms only
+  @Get("/:bookingUid")
+  async getBooking(@Param("bookingUid") bookingUid: string): Promise<ApiResponse<unknown>> {
+    const { bookingInfo } = await getBookingInfo(bookingUid);
```

Comment:
I see that "getBookingInfo" consists of "prisma.booking.findFirst" call from "apps/web/modules/bookings/views/bookings-single-view.getServerSideProps.tsx" - don't we also need to perform the rest of the operations e.g. eventType that is returned to frontend is transformed into final form in the get server side props:
```
const eventType = {
    ...eventTypeRaw,
    periodStartDate: eventTypeRaw.periodStartDate?.toString() ?? null,
    periodEndDate: eventTypeRaw.periodEndDate?.toString() ?? null,
    metadata: EventTypeMetaDataSchema.parse(eventTypeRaw.metadata),
    recurringEvent: parseRecurringEvent(eventTypeRaw.recurringEvent),
    customInputs: customInputSchema.array().parse(eventTypeRaw.customInputs),
  };
```

Reply:
I think this is done because next.js getServerSideProps can't serialize dates and other data structures